exo-go/src/docker_setup/index.go

```diff
@@ -0,0 +1,138 @@
+		builtDependency := appDependencyHelpers.Build(dependency, dockerSetup.AppConfig, dockerSetup.AppDir)
+		for variable, value := range builtDependency.GetServiceEnvVariables() {
+			result[variable] = value
+		}
+	}
+	for _, dependency := range dockerSetup.ServiceConfig.Dependencies {
+		result[strings.ToUpper(dependency.Name)] = dependency.Name
+	}
+	return result
+}
+
+func (dockerSetup *DockerSetup) getDockerLinks() []string {
+	result := []string{}
+	for _, dependency := range dockerSetup.ServiceConfig.Dependencies {
+		result = append(result, fmt.Sprintf("%s%s:%s", dependency.Name, dependency.Version, dependency.Name))
+	}
+	return result
+}
+
+func (dockerSetup *DockerSetup) getExternalServiceDockerConfigs() (map[string]types.DockerConfig, error) {
+	result := map[string]types.DockerConfig{}
+	renderedVolumes, err := dockerSetup.getRenderedVolumes(dockerSetup.ServiceConfig.Docker.Volumes)
+	if err != nil {
+		return result, err
+	}
+	result[dockerSetup.Role] = types.DockerConfig{
+		Image:         dockerSetup.ServiceData.DockerImage,
+		ContainerName: dockerSetup.Role,
+		Ports:         dockerSetup.ServiceConfig.Docker.Ports,
+		Environment:   util.JoinMaps(dockerSetup.ServiceConfig.Docker.Environment, dockerSetup.getDockerEnvVars()),
+		Volumes:       renderedVolumes,
+		DependsOn:     serviceConfigHelpers.GetServiceDependencies(dockerSetup.ServiceConfig, dockerSetup.AppConfig),
+	}
+	return result, nil
+}
+
+func (dockerSetup *DockerSetup) getInternalServiceDockerConfigs() (map[string]types.DockerConfig, error) {
+	result := map[string]types.DockerConfig{}
+	result[dockerSetup.Role] = types.DockerConfig{
+		Build:         path.Join("..", dockerSetup.ServiceData.Location),
+		ContainerName: dockerSetup.Role,
+		Command:       dockerSetup.ServiceConfig.Startup["command"],
+		Ports:         dockerSetup.ServiceConfig.Docker.Ports,
+		Links:         dockerSetup.getDockerLinks(),
+		Environment:   dockerSetup.getDockerEnvVars(),
+		DependsOn:     serviceConfigHelpers.GetServiceDependencies(dockerSetup.ServiceConfig, dockerSetup.AppConfig),
+	}
+	dependencyDockerConfigs, err := dockerSetup.getServiceDependenciesDockerConfigs()
+	if err != nil {
+		return result, err
+	}
+	return joinDockerConfigMaps(result, dependencyDockerConfigs), nil
+}
+
+func (dockerSetup *DockerSetup) getRenderedVolumes(volumes []string) ([]string, error) {
+	homeDir, err := osHelpers.GetUserHomeDir()
+	if err != nil {
+		return []string{}, err
+	}
+	dataPath := path.Join(homeDir, ".exosphere", dockerSetup.AppConfig.Name, dockerSetup.Role, "data")
+	renderedVolumes := []string{}
+	if err := os.MkdirAll(dataPath, 0777); err != nil { //nolint gas
+		return renderedVolumes, errors.Wrap(err, "Failed to create the necessary directories for the volumes")
+	}
+	for _, volume := range volumes {
+		renderedVolumes = append(renderedVolumes, strings.Replace(volume, "{{EXO_DATA_PATH}}", dataPath, -1))
+	}
+	return renderedVolumes, nil
```

Comment:
This is very similar to the getRenderedVolums in the first PR. I think we can create a helper to dry this up